### PR TITLE
Math signatures of elementwise and non-elementwise operators for multiplication and division [WIP]

### PR DESCRIPTION
**Pull request by adamhaber**
## Release notes

Fixed inconsistent use of elementwise and non-elementwise operators for multiplication and division

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by adamhaber** [comment]
@rok-cesnovar what would be the best place to add tests for these changes?

**Comment by rok-cesnovar** [comment]
I think just expand https://github.com/stan-dev/stanc3/blob/master/test/integration/good/function-signatures/math/matrix/elementwise_products.stan 

You can also place the scalar / matrix, matrix / scalar here as these are also technically elementwise operations.

File: src/middle/Stan_math_signatures.ml [hunk 1]
```diff
@@ -871,6 +871,7 @@ let () =
   add_unqualified ("divide", ReturnType UVector, [UVector; UReal]) ;
   add_unqualified ("divide", ReturnType URowVector, [URowVector; UReal]) ;
   add_unqualified ("divide", ReturnType UMatrix, [UMatrix; UReal]) ;
+  add_unqualified ("divide", ReturnType UMatrix, [UReal; UMatrix]) ;
```
Comment (rok-cesnovar) [inline src/middle/Stan_math_signatures.ml:874]:
```suggestion
  add_unqualified ("divide", ReturnType UVector, [UReal; UVector]) ;
  add_unqualified ("divide", ReturnType URowVector, [UReal; URowVector]) ;
  add_unqualified ("divide", ReturnType UMatrix, [UReal; UMatrix]) ;
```

File: src/middle/Stan_math_signatures.ml [hunk 2]
```diff
@@ -905,6 +906,8 @@ let () =
   add_unqualified
     ("elt_multiply", ReturnType URowVector, [URowVector; URowVector]) ;
   add_unqualified ("elt_multiply", ReturnType UMatrix, [UMatrix; UMatrix]) ;
+  add_unqualified ("elt_multiply", ReturnType UMatrix, [UMatrix; UReal]) ;
+  add_unqualified ("elt_multiply", ReturnType UMatrix, [UReal; UMatrix]) ;
```
Comment (rok-cesnovar) [inline src/middle/Stan_math_signatures.ml:910]: These will also work with row_vector and vector (matrix, vector and row_vector are eigen types so if it works for matrices it also automatically works for vectors/row_vectors. See the suggestion for divide.

**Comment by adamhaber** [comment]
@rok-cesnovar this should be ready for review now.

**Review by rok-cesnovar — APPROVED** [review]
Thanks Adam! Looks great.

**Comment by rok-cesnovar** [comment]
I had a look before, but got distracted and did not click approve. Thanks for the quick turnaround!

**Pull request merged**